### PR TITLE
fixed tests to work with new strats

### PR DESCRIPTION
**Pull request by timbotro**
## This PR contains: 
- Refactoring the test files back to the old format (sorry @bigwampa but it's neater this way)
- Adding updates to the test logic so it'll deploy a new strategy and config it if `strategyAddr` is supplied as empty string
- Test logic will deleverage to min existing strategy before deploying new one, `!redeem` gets thrown otherwise
- I've learnt the hard way that it's impossible to deploy new globes, they kept failing with `globe` reason
- Fixes to the static contract addresses for the USDC.e strategy contract files
- Fixes to helper insert function to support LP tokens
- Additions to static data files

N.B. Latest run is failing on DAI,USDC and WBTC on the same section, `withdrawAll()`:

image

You can see where i've put log lines in `generic-test.js` to help debug, but it's reverting without reason when test calls `await globeContract.connect(walletSigner).withdrawAll();` What's odd to me is that it's only failing for some assets and not all. Might be a clue as to why it's failing!

Running these tests against the old (the already deployed) contracts don't throw these errors.

---------------------
## Testing notes
Note that you may get out of memory issues when running this, because the default heapsize is 512. You can increase it to larger with `export NODE_OPTIONS=--max_old_space_size=4096` . This is unsurprising as we do have to load quite a lot events into memory.

You should be able to run this by cloning the env example file e.g. `cp .env.example .env` and running `npx hardhat test` . Note that for debugging you can actually run a separate node and run tests and connect a console to that. Happy to give a demo or record a screen vid next time of how i do my testing.

```
// Terminal1
npx hardhat node

// Terminal2
npx hardhat test --network localhost
// or "npx hardhat test --network localhost test/FoldingStrats/benqi-usdc.js"  if youre trying to just run a single test
npx hardhat console --network localhost
```

With the console attached you can do all sorts of interactive queries if you like:
```
let walletAddr = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266";
let assetAddr  = "0x50b7545627a5162F82A992c33b87aDc75187B218";
let walletSigner = ethers.provider.getSigner(walletAddr);
let assetContract = await ethers.getContractAt("ERC20",assetAddr,walletSigner);
await assetContract.balanceOf(walletAddr);
```

### CODE REVIEW

File: contracts/strategies/benqi/strategy-benqi-usdce.sol [hunk 1]
```diff
@@ -5,8 +5,8 @@ import "../strategy-qi-farm-base.sol";
 
 contract StrategyBenqiUsdc is StrategyQiFarmBase {
     
-    address public constant usdc = 0x5947BB275c521040051D82396192181b413227A3; //qideposit token
-    address public constant qiusdc = 0x4e9f683A27a6BdAD3FC2764003759277e93696e6; //lending receipt token
+    address public constant usdc = 0xA7D7079b0FEaD91F3e65f86E8915Cb59c1a4C664; //qideposit token
+    address public constant qiusdc = 0xBEb5d47A3f720Ec0a390d04b4d41ED7d9688bC7F; //lending receipt token
```
Comment (timbotro) [inline contracts/strategies/benqi/strategy-benqi-usdce.sol:10]: I'm glad this was caught!!!

**Comment by timbotro** [comment]
I've updated this PR to start working out the performance fees. Can confirm with the folding strats that setting performance fee will indeed increase treasury amount, but the amounts aren't quite what i was expecting for 10%:
image

image

Also are SNOBs only purchased for the LP pairs? I don't mention of it being converted in `strategy-qi-farm-base.sol`
